File: backend/schemas/artifact.py
"""
Artifact schemas for bug/feature tracking.
"""

from pydantic import BaseModel
from typing import Optional
from datetime import datetime


class ArtifactResponse(BaseModel):
    """Full artifact response."""
    id: int
    title: str
    description: Optional[str] = None
    artifact_type: str  # "bug" | "feature"
    status: str         # "open" | "in_progress" | "closed"
    created_by: int
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

File: frontend/src/types/artifact.ts
/**
 * Artifact types for bug/feature tracking.
 *
 * Mirrors backend schemas/artifact.py for easy cross-reference.
 */

export interface ArtifactCategory {
  id: number;
  name: string;
  created_at: string;
}

export interface Artifact {
  id: number;
  title: string;
  description: string | null;
  artifact_type: string;  // "bug" | "feature" | "task"
  status: string;         // "new" | "open" | "in_progress" | "icebox" | "closed"
  priority: string | null; // "urgent" | "high" | "medium" | "low"
  area: string | null;     // functional area (login_auth, streams, etc.)
  category: string | null;
  created_by: number;
  created_at: string;
  updated_at: string;
}
